### ISSUE TITLE
Make HorizontalLayout's alignment BASELINE by default

### ISSUE THREAD

**Issue by marcushellberg** [vaadin/flow-components#4756]
### Describe your motivation

Almost every time I use a HorizontalLayout, I end up with something that looks like this: 

<img width="337" alt="Screenshot 2023-03-01 at 16 49 08" src="https://user-images.githubusercontent.com/883698/222302340-e98183c5-8687-488f-8161-83efb3be82cc.png">

It means I have to save the layout to a variable and change the default alignment before adding it to a layout, increasing verbosity and adding clutter to the code. It is likely a source of frustration for beginners as things don't work as expected.

### Describe the solution you'd like

I would like the default alignment to be `baseline` so that the layout is more likely correct without additional configuration. 

### Describe alternatives you've considered

Having a fluent API for the layout so I can set the alignment without having to separately assign it to a variable. Something like: 

```java
add(
    new H1("Title!"), 
    new HorizontalLayout(input, button).setVerticalAlignment(Alignment.BASELINE)
)
```

**Comment by mstahv** [comment]
I agree that the default is not correct. If I'm not mistaken, I tend to more often find myself using CENTER though:

    layout.setAlignItems(FlexComponent.Alignment.CENTER)

**Comment by rolfsmeds** [comment]
Agree that the default, normal, is absolutely terrible, not only because it's rarely what you want and expect, but also because it [behaves differently in different situations](https://devdocs.io/css/align-items):

> The effect of this keyword is dependent of the layout mode we are in:
> 
> - In absolutely-positioned layouts, the keyword behaves like start on replaced absolutely-positioned boxes, and as stretch on all other absolutely-positioned boxes.
> - In static position of absolutely-positioned layouts, the keyword behaves as stretch.
> - For flex items, the keyword behaves as stretch.
> - For grid items, this keyword leads to a behavior similar to the one of stretch, except for boxes with an aspect ratio or an intrinsic sizes where it behaves like start.
> - The property doesn't apply to block-level boxes, and to table cells.

The challenge of course is every single existing Vaadin application out there, in which every single HL (and possible VL if we want to change that too) will be affected (admittedly in some layout instances won't be noticeable due to the configuration of the layout or the items in it, but in many it will).

So considering that

1. This is too big of a breaking change to ship in a minor in my opinion
2. Opting out of this change needs to be very easy

Opt-out could of course be achieved simply by adding a CSS block to a global stylesheet:

```
vaadin-horizonta-layout {
  align-items: normal;
}
```

It could also be provided through a CSS property like `--vaadin-horizontal-layout-default-vertical-alignment`, which could be set to `normal` to revert the change.
